pytensor/tensor/extra_ops.py

```diff
@@ -2018,6 +2018,42 @@ def broadcast_with_others(a, others):
     return brodacasted_vars
 
 
+def concat_with_broadcast(tensor_list, dim=0):
+    """
+    Concatenate a list of tensors, broadcasting the non-concatenated dimensions to align.
+    """
+    dim = dim if dim > 0 else tensor_list[0].ndim + dim
+    non_concat_shape = [None] * tensor_list[0].ndim
+
+    # If all inputs have the same broadcastable dim, it's not really a broadcast
+    all_bcast = [
```

Comment:
Set broadcast shape to all 1 (except axis), then you don't need the all_bcast logic? If you never overwrite the shape it was because everything was bcast on that axis